packages/lesswrong/components/posts/PostsPage/PostsPageActions.jsx

```diff
@@ -22,26 +26,50 @@ const showPostActions = (currentUser, post) => {
     Posts.canEdit(currentUser, post)
 }
 
-const PostsPageActions = ({classes, post, setMenuAnchor, anchorEl, currentUser}) => {
-  if (showPostActions(currentUser, post)) {
-    return <span>
-      <MoreHorizIcon className={classes.icon} onClick={e => setMenuAnchor(e.target)}/>
-      <Menu
-        anchorEl={anchorEl}
-        open={Boolean(anchorEl)}
-        onClose={() => setMenuAnchor(null)}
-      >
-        <Components.PostActions Container={MenuItem} post={post}/>
-      </Menu>
-    </span>
-  } else {
-    return null
+class PostsPageActions extends PureComponent {
+  state = { anchorEl: null }
+
+  handleClick = (e) => {
+    if (this.state.anchorEl) {
+      this.setState((prevState) => ({anchorEl: null}))
+    } else {
+      this.setState({anchorEl: e.target})
+    }
+  }
+
+  handleClose = (e) => {
+    // TODO – figure out why the prevent / stop thingies aren't working
+    this.setState({anchorEl: null})
+  }
+
+  render() {
+    const { classes, post, currentUser, vertical } = this.props 
+    const { anchorEl } = this.state 
+
+    if (showPostActions(currentUser, post)) {
```

Comment:
I would make this into an inline Boolean that's constructed at runtime, instead of a function

Reply:
Though I guess it's 4 lines or so, so abstracting that seems actually decent.